script/build.ts

```diff
@@ -131,21 +130,19 @@ function packageApp() {
     )
   }
 
-  const toPackageArch = (targetArch: string | undefined): OfficialArch => {
-    if (targetArch === undefined) {
-      targetArch = os.arch()
+  const getPackageArch = (): OfficialArch => {
+    const arch = process.env.npm_config_arch || process.arch
+
+    if (arch === 'arm64' || arch === 'x64' || arch === 'armv7l') {
+      return arch
     }
 
-    if (
-      targetArch === 'arm64' ||
-      targetArch === 'x64' ||
-      targetArch === 'armv7l'
-    ) {
-      return targetArch
+    if (arch === 'arm') {
+      return 'armv7l'
     }
 
     throw new Error(
-      `Building Desktop for architecture '${targetArch}' is not supported`
+      `Building Desktop for architecture '${arch}' is not supported`
```

Comment:
it would also be nice if we print out the supported architectures here
`x64` `arm64` and `arm` that way if the user (or a dev) hits this case they know what they did wrong

Reply:
9ac767e25fec0f16de3fd45ab4c576a809aa6ecb